src/OpenSage.Game/Data/Apt/ImageMap.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using OpenSage.Mathematics;
+
+namespace OpenSage.Data.Apt
+{
+
+    public interface ImageAssignment
+    {
+        bool HasBounds { get; }
+        int TextureId { get; }
+    }
+
+    public struct DirectAssignment : ImageAssignment
```

Comment:
This should probably be a `class`, along with the other assignment types. Using a struct that implements an interface involves boxing, which is relatively slow.